cpp/src/ray/test/api_test.cc

```diff
@@ -0,0 +1,152 @@
+
+#include <gtest/gtest.h>
+#include <ray/api.h>
+#include <future>
+#include <thread>
+
+using namespace ray::api;
+
+int Return1() { return 1; }
+int Plus1(int x) { return x + 1; }
+
+int Plus(int x, int y) { return x + y; }
+
+class Counter {
+ public:
+  int count;
+
+  MSGPACK_DEFINE(count);
+
+  Counter() { count = 0; }
+
+  static Counter *FactoryCreate() {
+    Counter *counter = new Counter();
+    return counter;
+  }
+
+  int Plus1(int x) { return x + 1; }
+
+  int Plus(int x, int y) { return x + y; }
+
+  int Add(int x) {
+    count += x;
+    return count;
+  }
+
+  static int Plus1S(int x) { return x + 1; }
+
+  static int PlusS(int x, int y) { return x + y; }
```

Comment:
Does the tailing `S` mean static? Looks a bit weird, I would just name it `Plus1`.